schema_based_on_diagram.sql

```diff
@@ -1 +1,45 @@
+CREATE TABLE treatments (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    type VARCHAR(30),
+    name VARCHAR(30),
+    PRIMARY KEY(id),
+);
+
+CREATE TABLE medical_histories (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    admitted_at TIMESTAMP,
+    patient_id INT,
+    status VARCHAR(30),
+    PRIMARY KEY(id),
+    FOREIGN KEY(patient_id) REFERENCES patients(id)
+);
+
+CREATE TABLE patients(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  name VARCHAR(20),
+  date_of_birth DATE,
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE invoices(
+  id INT GENERATED ALWAYS AS IDENTITY,
+  total_amountsql DECIMAL,
+  generated_at TIMESTAMP,
+  generated_at TIMESTAMP,
+  medical_history_id INT
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE invoice_items (
+  id INT,
+  unite_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT,
+  PRIMARY KEY(id),
+  FOREIGN KEY(invoice_id) REFERENCES invoices(id),
+  FOREIGN KEY(treatment_id) REFERENCES treatments(id),
+);
+
 
```

Comment:
- [ ] Finally please don't forget that all FOREIGN KEYS should have indexes created:

* `CREATE INDEX ON medical_histories (patient_id);`
* `CREATE INDEX ON invoices (medical_history_id);`
* `CREATE INDEX ON invoice_items (invoice_id);`
* `CREATE INDEX ON invoice_items (treatment_id);`
* `CREATE INDEX ON medical_histories_has_treatments (medical_history_id);`
* `CREATE INDEX ON medical_histories_has_treatments (treatment_id);`